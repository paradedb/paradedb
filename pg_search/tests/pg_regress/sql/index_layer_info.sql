\i common/common_setup.sql

CALL paradedb.create_bm25_test_table(
  schema_name => 'public',
  table_name => 'mock_items_1'
);

CALL paradedb.create_bm25_test_table(
  schema_name => 'public',
  table_name => 'mock_items_2'
);

CREATE INDEX mock_items_1_idx ON mock_items_1
USING bm25 (id, description, category)
WITH (key_field='id');

CREATE INDEX mock_items_2_idx ON mock_items_2
USING bm25 (id, description, category)
WITH (key_field='id');

SELECT relname, layer_size FROM pdb.index_layer_info WHERE relname = 'mock_items_1_idx' OR relname = 'mock_items_2_idx';
SELECT * FROM paradedb.combined_layer_sizes('mock_items_1_idx');
SELECT * FROM paradedb.combined_layer_sizes('mock_items_2_idx');

ALTER INDEX mock_items_1_idx SET (layer_sizes = '0');
ALTER INDEX mock_items_1_idx SET (background_layer_sizes = '10kb, 100kb, 1mb, 100mb');

SELECT relname, layer_size FROM pdb.index_layer_info WHERE relname = 'mock_items_1_idx' OR relname = 'mock_items_2_idx';
SELECT * FROM paradedb.combined_layer_sizes('mock_items_1_idx');

ALTER INDEX mock_items_1_idx SET (layer_sizes = '10kb, 100kb');
ALTER INDEX mock_items_1_idx SET (background_layer_sizes = '10kb, 100kb, 1mb, 100mb, 1gb');

SELECT relname, layer_size FROM pdb.index_layer_info WHERE relname = 'mock_items_1_idx' OR relname = 'mock_items_2_idx';
SELECT * FROM paradedb.combined_layer_sizes('mock_items_1_idx');

DROP TABLE mock_items_1;
DROP TABLE mock_items_2;
