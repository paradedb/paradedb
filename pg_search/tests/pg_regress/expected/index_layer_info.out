\i common/common_setup.sql
CREATE EXTENSION IF NOT EXISTS pg_search;
-- Disable parallel workers to avoid differences in plans
SET max_parallel_workers_per_gather = 0;
SET enable_indexscan to OFF;
SET paradedb.enable_mixed_fast_field_exec = true;
CALL paradedb.create_bm25_test_table(
  schema_name => 'public',
  table_name => 'mock_items_1'
);
CALL paradedb.create_bm25_test_table(
  schema_name => 'public',
  table_name => 'mock_items_2'
);
CREATE INDEX mock_items_1_idx ON mock_items_1
USING bm25 (id, description, category)
WITH (key_field='id');
CREATE INDEX mock_items_2_idx ON mock_items_2
USING bm25 (id, description, category)
WITH (key_field='id');
SELECT relname, layer_size FROM pdb.index_layer_info WHERE relname = 'mock_items_1_idx' OR relname = 'mock_items_2_idx';
     relname      |    layer_size     
------------------+-------------------
 mock_items_1_idx | 977 GB..
 mock_items_1_idx | 98 GB..977 GB
 mock_items_1_idx | 10000 MB..98 GB
 mock_items_1_idx | 1000 MB..10000 MB
 mock_items_1_idx | 100 MB..1000 MB
 mock_items_1_idx | 10 MB..100 MB
 mock_items_1_idx | 1024 kB..10 MB
 mock_items_1_idx | 100 kB..1024 kB
 mock_items_1_idx | 10 kB..100 kB
 mock_items_1_idx | ..10 kB
 mock_items_2_idx | 977 GB..
 mock_items_2_idx | 98 GB..977 GB
 mock_items_2_idx | 10000 MB..98 GB
 mock_items_2_idx | 1000 MB..10000 MB
 mock_items_2_idx | 100 MB..1000 MB
 mock_items_2_idx | 10 MB..100 MB
 mock_items_2_idx | 1024 kB..10 MB
 mock_items_2_idx | 100 kB..1024 kB
 mock_items_2_idx | 10 kB..100 kB
 mock_items_2_idx | ..10 kB
(20 rows)

SELECT * FROM paradedb.combined_layer_sizes('mock_items_1_idx');
                                    combined_layer_sizes                                     
---------------------------------------------------------------------------------------------
 {10240,102400,1048576,10485760,104857600,1048576000,10485760000,104857600000,1048576000000}
(1 row)

SELECT * FROM paradedb.combined_layer_sizes('mock_items_2_idx');
                                    combined_layer_sizes                                     
---------------------------------------------------------------------------------------------
 {10240,102400,1048576,10485760,104857600,1048576000,10485760000,104857600000,1048576000000}
(1 row)

ALTER INDEX mock_items_1_idx SET (layer_sizes = '0');
ALTER INDEX mock_items_1_idx SET (background_layer_sizes = '10kb, 100kb, 1mb, 100mb');
SELECT relname, layer_size FROM pdb.index_layer_info WHERE relname = 'mock_items_1_idx' OR relname = 'mock_items_2_idx';
     relname      |    layer_size     
------------------+-------------------
 mock_items_1_idx | 100 MB..
 mock_items_1_idx | 1024 kB..100 MB
 mock_items_1_idx | 100 kB..1024 kB
 mock_items_1_idx | 10 kB..100 kB
 mock_items_1_idx | ..10 kB
 mock_items_2_idx | 977 GB..
 mock_items_2_idx | 98 GB..977 GB
 mock_items_2_idx | 10000 MB..98 GB
 mock_items_2_idx | 1000 MB..10000 MB
 mock_items_2_idx | 100 MB..1000 MB
 mock_items_2_idx | 10 MB..100 MB
 mock_items_2_idx | 1024 kB..10 MB
 mock_items_2_idx | 100 kB..1024 kB
 mock_items_2_idx | 10 kB..100 kB
 mock_items_2_idx | ..10 kB
(15 rows)

SELECT * FROM paradedb.combined_layer_sizes('mock_items_1_idx');
       combined_layer_sizes       
----------------------------------
 {10240,102400,1048576,104857600}
(1 row)

ALTER INDEX mock_items_1_idx SET (layer_sizes = '10kb, 100kb');
ALTER INDEX mock_items_1_idx SET (background_layer_sizes = '10kb, 100kb, 1mb, 100mb, 1gb');
SELECT relname, layer_size FROM pdb.index_layer_info WHERE relname = 'mock_items_1_idx' OR relname = 'mock_items_2_idx';
     relname      |    layer_size     
------------------+-------------------
 mock_items_1_idx | 1024 MB..
 mock_items_1_idx | 100 MB..1024 MB
 mock_items_1_idx | 1024 kB..100 MB
 mock_items_1_idx | 100 kB..1024 kB
 mock_items_1_idx | 10 kB..100 kB
 mock_items_1_idx | ..10 kB
 mock_items_2_idx | 977 GB..
 mock_items_2_idx | 98 GB..977 GB
 mock_items_2_idx | 10000 MB..98 GB
 mock_items_2_idx | 1000 MB..10000 MB
 mock_items_2_idx | 100 MB..1000 MB
 mock_items_2_idx | 10 MB..100 MB
 mock_items_2_idx | 1024 kB..10 MB
 mock_items_2_idx | 100 kB..1024 kB
 mock_items_2_idx | 10 kB..100 kB
 mock_items_2_idx | ..10 kB
(16 rows)

SELECT * FROM paradedb.combined_layer_sizes('mock_items_1_idx');
            combined_layer_sizes             
---------------------------------------------
 {10240,102400,1048576,104857600,1073741824}
(1 row)

DROP TABLE mock_items_1;
DROP TABLE mock_items_2;
